src/storage/src/table/cell_based_table.rs

```diff
@@ -456,3 +464,69 @@ impl<S: StateStore> TableIter for CellBasedTableRowIter<S> {
         }
     }
 }
+
+/// "`CellBasedTableStreamingIter`" used for streaming executor, and will not wait epoch
+pub struct CellBasedTableStreamingIter<S: StateStore> {
+    /// An iterator that returns raw bytes from storage.
+    iter: StripPrefixIterator<S::Iter>,
+    /// Cell-based row deserializer
+    cell_based_row_deserializer: CellBasedRowDeserializer,
+    pub cell_based_item: Option<(Vec<u8>, Row)>,
+}
+
+impl<S: StateStore> CellBasedTableStreamingIter<S> {
+    pub async fn new(
+        keyspace: &Keyspace<S>,
+        table_descs: &[ColumnDesc],
+        epoch: u64,
+    ) -> StorageResult<Self> {
+        let cell_based_row_deserializer = CellBasedRowDeserializer::new(table_descs.to_vec());
+        let iter = keyspace.iter(epoch).await?;
+        let iter = Self {
+            iter,
+            cell_based_row_deserializer,
+            cell_based_item: None,
+        };
+        Ok(iter)
+    }
+
+    pub async fn init_cell_based_item(&mut self) -> StorageResult<()> {
+        if self.cell_based_item.is_none() {
+            self.cell_based_item = self.next().await?;
+        }
+        Ok(())
+    }
+
+    pub async fn update_current_item(&mut self) -> StorageResult<()> {
+        self.cell_based_item = self.next().await?;
+        Ok(())
+    }
```

Comment:
Are these methods used for making this iterator peekable? I still suggest making `Peekable` a wrapper or simply using `StreamExt::peekable` after we refactor this with async stream.

If we call `update_current_item` without checking `cell_based_item` then, we may forget about this peeked item, this is strange. You may check whether `self.cell_based_item` is `Some` in the `next` method like how `std::iter::Peekable` is implemented.